lib/modules/platform/bitbucket-server/index.ts

```diff
@@ -373,11 +374,13 @@ export async function getBranchPr(branchName: string): Promise<BbsPr | null> {
 }
 
 // istanbul ignore next
-export async function refreshPr(number: number): Promise<void> {
+export async function refreshPr({
```

Comment:
[Renovate's `platformAutomerge` docs](https://docs.renovatebot.com/configuration-options/#platformautomerge) should be updated, once the code is done.

Reply:
I'm struggling a bit to come up with a way to phrase this that's up to the high quality bar set by the rest of the docs, but I made an attempt in the last push!

Reply:
I just noticed that the GitLab platform calls its `tryPrAutomerge` during both `createPr` and `updatePr`. Perhaps the sentence "Renovate tries platform-native automerge only when it initially creates the PR." in the doc was already incomplete or out-of-date?

https://github.com/renovatebot/renovate/blob/0b3b56adb555ae993a8f89ea721c5dfa36c4b259/lib/modules/platform/gitlab/index.ts#L818

Reply:
Now realizing the above, I wonder if I should rework this whole change for consistency with the gitlab behavior, and either:
1. add a call to `tryPrAutomerge` in the github platform's `updatePr`, or 
2. refactor gitlab to have and use the new `reattemptPlatformAutomerge` method instead.

Reply:
@HonkingGoose @rarkins @viceice Any thoughts on the above? I'm happy to keep moving as-is or change my approach

Reply:
I'll let a maintainer respond. 🙂

Reply:
I have lost track of how/why gitlab's approach is like it is. It's possible that the docs are inaccurate.

Isn't updatePr() only called if the PR title or body needs changing? i.e. a force push which does not require a change of PR title or body won't result in automerge being re-enabled?

Reply:
From [some digging just now](https://github.com/renovatebot/renovate/blob/99467af092fb621be56f515e4b7aed7d6ccaf51e/lib/workers/repository/update/pr/index.ts#L347-L351), it looks like we'll get to `updatePr` if the base branch, title, or body change, so
> i.e. a force push which does not require a change of PR title or body won't result in automerge being re-enabled?

does seem like a pretty accurate description. Still sounds inconsistent with the docs "only when it initially creates the PR" wording in GitLab's case, but I don't have personal experience with GitLab

Reply:
I see the GitLab retry automerge was added quite long ago by @viceice: https://github.com/renovatebot/renovate/pull/7871

@straub thanks for pointing this out and apologies that it's holding up this PR, but I think you're right that we should resolve this inconsistency.

When designing this feature, we postured that re-enabling automerge after commits was the safest/most conservative approach, because we want to avoid flip/flopping if a user tries disabling automerge and Renovate keeps turning it back on.

Meanwhile we rediscover that GitLab apparently re-enables automerge in the most common scenario where the PR title or body changes, and I guess we haven't heard complaints from users.

I think that the force push approach seems best, and it's mostly a superset of the updatePr() use cases. There are cases such as a git conflict where there will be a force push without need to update the PR content. Meanwhile are there cases where the PR needs updating without a force push where automerge needs re-enabling? I can't think of any.

Reply:
Sorry, somehow I missed this response. I'll update GitLab to use `reattemptPlatformAutomerge` 👍

Reply:
@rarkins, I've got both `github` and `gitlab` platforms using `reattemptPlatformAutomerge` now, and I've updated the docs to reflect that. Thanks for your patience!